contracts/pol/CarbonPOL.sol

```diff
@@ -0,0 +1,303 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+import { IVersioned } from "../utility/interfaces/IVersioned.sol";
+import { ICarbonPOL } from "./interfaces/ICarbonPOL.sol";
+import { Upgradeable } from "../utility/Upgradeable.sol";
+import { Token } from "../token/Token.sol";
+import { Utils } from "../utility/Utils.sol";
+import { MathEx } from "../utility/MathEx.sol";
+import { ExpDecayMath } from "../utility/ExpDecayMath.sol";
+import { MAX_GAP, PPM_RESOLUTION } from "../utility/Constants.sol";
+
+/**
+ * @notice CarbonPOL contract
+ */
+contract CarbonPOL is ICarbonPOL, Upgradeable, ReentrancyGuardUpgradeable, Utils {
```

Comment:
BTW, any reason why we have stopped using `onlyProxyDelegate` in this (and other) contracts?

Reply:
I think there was an extra step necessary (upgrade) to set the proxy address for the contract, making it automatically start with version 2 to function normally.

Reply:
What do you mean? `onlyProxyDelegate` is only used for IP protection.

Reply:
this was only part of the CarbonController contract

Reply:
Ah, gotcha. Resolving.